packages/hyperion-autologging/src/ALTriggerFlowlet.ts

```diff
@@ -86,6 +101,38 @@ export function init(options: InitOptions) {
     setSurfaceTriggerFlowlet(event.surface, event.triggerFlowlet);
   });
 
+
+  const IS_TRIGGER_FLOWLET_SETUP_PROP = 'isTriggerFlowletSetup';
+  IEventTarget.addEventListener.onArgsObserverAdd(function (this, eventType, _callback) {
+    if (!ALUIEVentGroupPublishers.isSupported(eventType)) {
+      return;
+    }
+
+    if (!isTrackedEvent(eventType) && this instanceof Element && !getVirtualPropertyValue(this, IS_TRIGGER_FLOWLET_SETUP_PROP)) {
+      setVirtualPropertyValue(this, IS_TRIGGER_FLOWLET_SETUP_PROP, true);
+      /**
+       * We could potentially intercept the callback itself and add the following
+       * but that could add too much overhead to every callback.
+       */
+      // const topTriggerFlowlet = flowletManager.top()?.data.triggerFlowlet;
+      IEventTarget.addEventListener.getOriginal().call(
```

Comment:
For my understanding what is this doing exactly?  We are inside of an `onArgsObserverAdd` for the same `addEventListener` that we are intercepting but then doing a call on the original function that was intercepted?

Reply:
I'm glad you noticed this! :).
So, basically we catch application code calling `addEventListener` here to add some event. We then want to do one extra event registration of our own, the very first time such event handler is installed. Here if we call the normal `addEventListener`, it will again bring us back to this very interception and we go into an infinite loop! 
We do know that the goal of this call back is to update the triggerFlowlet and nothing else matters. So, we can bypass all of the other logic by calling the original function. 

I'll add this comment to the code itself in an update.